File: repos/appTiendaUPN/appTiendaUPN/Models/Producto.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace appTiendaUPN.Models
{
    [Table("productos")]
    public class Producto
    {
        [Key]
        [Column("productoid")]
        public int ProductoId { get; set; }

        [Required]
        [MaxLength(200)]
        [Column("nombre")]
        public string Nombre { get; set; } = string.Empty;

        [Column("descripcion")]
        public string? Descripcion { get; set; }

        [Required]
        [Column("precio", TypeName = "decimal(18,2)")]
        public decimal Precio { get; set; }

        [Column("precioanterior", TypeName = "decimal(18,2)")]
        public decimal? PrecioAnterior { get; set; }

        [Required]
        [Column("stock")]
        public int Stock { get; set; }

        [MaxLength(500)]
        [Column("imagenurl")]
        public string? ImagenUrl { get; set; }

        [Required]
        [Column("categoriaid")]
        public int CategoriaId { get; set; }

        [Column("estaactivo")]
        public bool EstaActivo { get; set; } = true;

        [Column("esoferta")]
        public bool EsOferta { get; set; } = false;

        [Column("fechacreacion")]
        public DateTime FechaCreacion { get; set; } = DateTime.UtcNow;

        // Relación con Categoría
        [ForeignKey("CategoriaId")]
        public Categoria? Categoria { get; set; }
    }
}

File: repos/Soluciones/appFinal_Backend/src/main/java/com/example/appfinal_backend/model/Producto.java
package com.example.appfinal_backend.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "productos")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Producto {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "productoid")
    private Integer productoId;
    
    @Column(name = "nombre", nullable = false, length = 200)
    private String nombre;
    
    @Column(name = "descripcion", columnDefinition = "TEXT")
    private String descripcion;
    
    @Column(name = "precio", nullable = false, precision = 18, scale = 2)
    private BigDecimal precio;
    
    @Column(name = "precioanterior", precision = 18, scale = 2)
    private BigDecimal precioAnterior;
    
    @Column(name = "stock", nullable = false)
    private Integer stock = 0;
    
    @Column(name = "imagenurl", length = 500)
    private String imagenUrl;
    
    @Column(name = "categoriaid", nullable = false)
    private Integer categoriaId;
    
    @Column(name = "estaactivo")
    private Boolean estaActivo = true;
    
    @Column(name = "esoferta")
    private Boolean esOferta = false;
    
    @CreationTimestamp
    @Column(name = "fechacreacion", updatable = false)
    private LocalDateTime fechaCreacion;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "categoriaid", insertable = false, updatable = false)
    private Categoria categoria;


    @Column(name = "mensaje", columnDefinition = "TEXT")
    private String mensaje;
}
